CONTRIBUTING.rst

```diff
@@ -0,0 +1,477 @@
+that the project uses. These do not install automatically when you
+install ``drizzlepac`` unless directly specified. To do this, while in
+the top level directory of ``drizzlepac`` on your my_feature branch:
+
+::
+
+   >> pip install -e ".[docs]"
+
+Now, with the correct documentation dependencies installed, you can
+attempt to build the documentation locally. To do this, enter into the
+``drizzlepac/docs`` subdirectory and do:
+
+::
+
+   >> make html
+
+If the documentation successfully builds, the output HTML files will be
+output in the ``_build/html`` subdirectory. You can open the main
+``index.html`` file in your browser and explore the full documentation
+pages just like the readthedocs site. If there were any errors or
+warnings, a traceback will be printed on your terminal. Small errors,
+like a typo in a link to another section, can cause frustrating errors
+so it is good practice to build the docs frequently when editing them.
+
+Writing and running unit tests
+------------------------------
+
+Unit tests are located in the ``tests`` directory and are separated by instrument,
+with additional separate directories for Hubble Advanced Products (HAP) and the 
+drizzle algorithm. These tests run the code
+on simplified datasets to make sure there are no breaking changes
+introduced. We aim to cover most lines of ``drizzlepac`` with unit
+test, so when adding code you will often need to write a new test or add
+to an existing test to ensure adequate coverage.
+
+Take a look around at the existing tests - many tests use a @pytest.fixture to 
+set up a common dataset. The test functions themselves set up a scenario to run a 
+function under certain conditions, and culminate in a set of assertions that need to
+pass (or fail if the test is marked as ``xfail``).
+
+The CI will run the unit tests on your branch when you open a pull
+request. They will re-run every time you push commits to this remote
+branch as well. Unit tests must pass on any changes you make, and if
+you're introducing new code that isn't covered by an existing test, you
+will need to write one. The ``codecoverage`` check that runs on the CI
+when you open a pull request will tell you if you've introduced any new
+lines that aren't covered by a test.
+
+You can also run unit tests locally, and you should do this periodically
+to test your code. To do this, you will need to install the optional
+dependencies needed for running tests.
+
+::
+
+   >> pip install -e ".[test]"
+
+This will install the optional 'test' dependencies specified in
+``pyproject.toml`` that don't install by default. The package ``pytest``
+is one of these and is what's used to run the tests. ``pytest`` searches
+through all the directories in your repository (underneath the directory
+from which it was invoked command line) and looks for any directories
+called 'test' or .py files with the word 'test' in the name. Functions
+in these files will be executed.
+
```

Comment:
Steve, Put the following text where you think it is the most appropriate and reword as you want. Further down in this write-up may be better for this discussion where you talk about flag options.  You will see hints of this in the JenkinsfileRT.

The tests which import hst_helpers from the ci_watson package (e.g., hap, acs, wfc3) need to have the environment variable TEST_BIGDATA defined for the appropriate location in Artifactory (e.g., TEST_BIGDATA=https://bytesalad.stsci.edu/artifactory/) or directory path on your local disk where the _input_ and _truth_ files area located.  You also need to use the flag, --bigdata, to have the FITS files downloaded into the testing area.

Some tests are also marked as "slow" and will only be executed when the "--slow" flag is included on the "pytest" command line.  These tests are typically NOT executed as part of the automatically invoked regression test suite due the amount of time needed to execute the test.

Reply:
I was under the impression that this document was more for a possible non-STScI contributor, whereas our internal notes are on the innerspace page. Can a non-STScI user run the --slow and --bigdata tests? My impression was that they couldn't, but I may be wrong.